### ISSUE TITLE
Stored Object: paper:arxiv.2609.04061

### ISSUE THREAD

**Issue by soyomarvaldezg** [soyomarvaldezg/papers-feed#50]
{
  "sourceId": "arxiv",
  "paperId": "2609.04061",
  "url": "https://arxiv.org/abs/2609.04061",
  "title": "When Models Edit Too Much: On the Fidelity of Minimal Code Edits",
  "authors": "Tongyao Zhu, Wei Hern Lim, Min-Yen Kan",
  "abstract": "Large language models (LLMs) are increasingly used to edit existing code, but correctness alone is not enough: useful repairs should also be minimal, reviewable, and faithful to the original implementation. We study over-editing, the tendency of a model to rewrite code beyond what is required to fix a bug. We construct an evaluation framework from 400 BigCodeBench problems by injecting controlled AST-level corruptions into reference solutions, giving each repair task a known minimal patch. Across frontier LLMs, over-editing is widespread even among strong models like GPT-5.5: high Pass@1 can coexist with unnecessarily large edits and added cognitive complexity. A preservation instruction substantially reduces this behavior, lowering average excess Levenshtein distance from 0.195 to 0.131, reducing added cognitive complexity by 26.6%, and increasing Pass@1 by 2.3 points. However, these gains do not simply follow from a larger reasoning budget or larger models. We next ask whether minimal editing can be learned directly during post-training. We observe that supervised fine-tuning overfits to seen corruption patterns, whereas reinforcement learning gives the best out-of-domain edit-fidelity and performance-retention trade-off. These results position edit fidelity as a distinct axis of code-repair quality and show that it can be measured and learned.",
  "timestamp": "2026-09-04T15:10:20.515Z",
  "publishedDate": "2026/09/03",
  "tags": [
    "Software Engineering (cs.SE)",
    "Artificial Intelligence (cs.AI)",
    "Computation and Language (cs.CL)"
  ],
  "rating": "novote",
  "sourceType": "url"
}

**Comment by soyomarvaldezg** [comment]
{
  "_data": {
    "sourceId": "arxiv",
    "paperId": "2609.04061",
    "url": "https://arxiv.org/abs/2609.04061",
    "title": "When Models Edit Too Much: On the Fidelity of Minimal Code Edits",
    "authors": "Tongyao Zhu, Wei Hern Lim, Min-Yen Kan",
    "abstract": "Large language models (LLMs) are increasingly used to edit existing code, but correctness alone is not enough: useful repairs should also be minimal, reviewable, and faithful to the original implementation. We study over-editing, the tendency of a model to rewrite code beyond what is required to fix a bug. We construct an evaluation framework from 400 BigCodeBench problems by injecting controlled AST-level corruptions into reference solutions, giving each repair task a known minimal patch. Across frontier LLMs, over-editing is widespread even among strong models like GPT-5.5: high Pass@1 can coexist with unnecessarily large edits and added cognitive complexity. A preservation instruction substantially reduces this behavior, lowering average excess Levenshtein distance from 0.195 to 0.131, reducing added cognitive complexity by 26.6%, and increasing Pass@1 by 2.3 points. However, these gains do not simply follow from a larger reasoning budget or larger models. We next ask whether minimal editing can be learned directly during post-training. We observe that supervised fine-tuning overfits to seen corruption patterns, whereas reinforcement learning gives the best out-of-domain edit-fidelity and performance-retention trade-off. These results position edit fidelity as a distinct axis of code-repair quality and show that it can be measured and learned.",
    "timestamp": "2026-09-04T15:10:20.515Z",
    "publishedDate": "2026/09/03",
    "tags": [
      "Software Engineering (cs.SE)",
      "Artificial Intelligence (cs.AI)",
      "Computation and Language (cs.CL)"
    ],
    "rating": "novote",
    "sourceType": "url"
  },
  "_meta": {
    "client_version": "0.11.1",
    "timestamp": "2026-09-04T15:10:21.030Z",
    "update_mode": "append",
    "issue_number": 50
  },
  "type": "initial_state"
}
